### PR TITLE
Update more tests

### PR DESCRIPTION
**Pull request by richsalz**
modes_internal_test, sslcorrupttest, v3nametest

### CODE REVIEW

**Comment by dot-asm** [comment]
Red cross from appveyour is relevant.

File: test/v3nametest.c [hunk 1]
```diff
@@ -238,118 +243,117 @@ static const struct set_name_fn name_fns[] = {
...
-static void run_cert(X509 *crt, const char *nameincert,
+static int run_cert(X509 *crt, const char *nameincert,
                      const struct set_name_fn *fn)
 {
     const char *const *pname = names;
-    while (*pname) {
+    int failed = 0;
+
+    for (; *pname != NULL; ++pname) {
         int samename = strcasecmp(nameincert, *pname) == 0;
```
Comment (levitte) [inline test/v3nametest.c]: You need to include `e_os.h` for this to be correctly defined on Windows

Comment (richsalz) [inline test/v3nametest.c]: oops.  updated commit pushed; let's see how travis does.

**Comment by richsalz** [comment]
f3ab6c1 in master, thanks!

**Pull request closed without merging**